packages/team-management/src/components/TeamsView/index.tsx

```diff
@@ -0,0 +1,188 @@
+import { ERROR_OCCURRED, TEAMS_GET, TEAM_PRACTITIONERS, URL_ADD_TEAM } from '../../constants';
+import Table, { TableData } from './Table';
+import './TeamsView.css';
+import { Spin } from 'antd';
+import { Link } from 'react-router-dom';
+import { Practitioner } from '../../ducks/practitioners';
+
+/** Register reducer */
+reducerRegistry.register(orgReducerName, organizationsReducer);
+
+/**
+ * Function to load selected Team for details
+ *
+ * @param {TableData} row data selected from the table
+ * @param {string} accessToken - access token
+ * @param {string} opensrpBaseURL - base url
+ * @param {Function} setDetail funtion to set detail to state
+ * @param {Function} setPractitionersList funtion to set detail to state
+ */
+export const loadSingleTeam = (
+  row: TableData,
+  accessToken: string,
+  opensrpBaseURL: string,
+  setDetail: (isLoading: string | Organization) => void,
+  setPractitionersList: (isLoading: string | Practitioner[]) => void
+): void => {
+  const serve = new OpenSRPService(
+    accessToken,
+    opensrpBaseURL,
+    TEAM_PRACTITIONERS + row.identifier
+  );
+  serve
+    .list()
+    .then((response: Practitioner[]) => {
+      setPractitionersList(response);
+      setDetail(row);
+    })
+    .catch(() => sendErrorNotification(ERROR_OCCURRED));
+};
+
+export interface Props {
+  opensrpBaseURL: string;
+}
+
+/** default component props */
+export const defaultProps = {
+  opensrpBaseURL: '',
+};
+
+/** Function which shows the list of all teams and there details
+ *
+ * @param {Object} props - TeamsView component props
+ * @returns {Function} returns team display
+ */
+export const TeamsView: React.FC<Props> = (props: Props) => {
+  const dispatch = useDispatch();
+  const accessToken = useSelector((state) => getAccessToken(state) as string);
+  const teamsArray = useSelector((state) => getOrganizationsArray(state));
+  const [detail, setDetail] = useState<TeamsDetailProps | null>(null);
+  const [practitionersList, setPractitionersList] = useState<Practitioner[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [value, setValue] = useState('');
+  const [filter, setfilterData] = useState<TableData[] | null>(null);
+  const { opensrpBaseURL } = props;
+
+  useEffect(() => {
+    if (isLoading) {
+      const serve = new OpenSRPService(accessToken, opensrpBaseURL, TEAMS_GET);
+      serve
+        .list()
+        .then((response: Organization[]) => {
+          dispatch(fetchOrganizationsAction(response));
+          setIsLoading(false);
+        })
+        .catch(() => sendErrorNotification(ERROR_OCCURRED));
+    }
+  });
+
+  const tableData: TableData[] = [];
+
+  if (teamsArray.length) {
+    teamsArray.forEach((team: Organization, i: number) => {
+      tableData.push({
+        key: i.toString(),
+        id: team.id,
+        name: team.name,
+        active: team.active,
+        identifier: team.identifier,
+      });
+    });
```

Comment:
lets use array.map instead of foreach

Reply:
We are not changing anything in `teamsArray` just looping it to store in `tableData` array, So in my opinion `forEach` is a better fit.

Reply:
https://codeburst.io/javascript-map-vs-foreach-f38111822c0f#:~:text=changing%20or%20altering%20data

this article might help a bit i just the coding scheme although its not required